crates/ruff_linter/src/rules/pyupgrade/rules/f_strings.rs

```diff
@@ -455,30 +453,28 @@ pub(crate) fn f_strings(checker: &mut Checker, call: &ast::ExprCall, summary: &F
 
     let mut contents = String::with_capacity(checker.locator().slice(call).len());
     let mut prev_end = call.start();
-    for (range, fstring) in patches {
-        contents.push_str(
-            checker
-                .locator()
-                .slice(TextRange::new(prev_end, range.start())),
-        );
-        contents.push_str(&fstring);
-        prev_end = range.end();
-    }
-
-    // If the remainder is non-empty, add it to the contents.
-    let rest = checker.locator().slice(TextRange::new(prev_end, end));
-    if !lexer::lex_starts_at(rest, Mode::Expression, prev_end)
-        .flatten()
-        .all(|(token, _)| match token {
-            Tok::Comment(_) | Tok::Newline | Tok::NonLogicalNewline | Tok::Indent | Tok::Dedent => {
-                true
+    for (range, conversion) in patches {
+        let fstring = match conversion {
+            FStringConversion::Convert(fstring) => Some(fstring),
+            FStringConversion::EmptyLiteral => None,
+            FStringConversion::Literal => {
+                // Convert escaped curly brackets e.g. `{{` to `{` in literal string parts
+                Some(curly_unescape(checker.locator().slice(range)).to_string())
             }
-            Tok::String { value, .. } => value.is_empty(),
-            _ => false,
-        })
```

Comment:
The main reason for this PR is to remove this little lexer usage which uses the owned data from `String` token.